### PR TITLE
Do not open atomic_file for mkdirs result.

### PR DESCRIPTION
**Pull request by themalkolm**
Looks like we open file here only for a side effect that we mkdir
all folders. It is quite silly, lets just mkdir without open/close
invocations.

Lets see what Travis thinks :D

### CODE REVIEW

**Comment by erikbern** [comment]
LGTM assuming it works

File: luigi/file.py [hunk 1]
```diff
@@ -81,14 +81,16 @@ def __init__(self, path=None, format=None, is_tmp=False):
         self.format = format
         self.is_tmp = is_tmp
 
+    def mkdirs(self):
```
Comment (Tarrasch) [inline luigi/file.py]: Really short method name. Make it describe what you do. Like `create_parent_directory` or whatever you think suits.

Comment (themalkolm) [inline luigi/file.py]: I think mkdirs is enough and it is on the same level of understanding as open,close,copy,move. Surely could rename to `makedirs` mimicking the actual os call `os.makedirs` if it helps.

Comment (Tarrasch) [inline luigi/file.py]: I didn't know `os.makedirs` was a thing. But since it it? Why don't you call it `makedirs` then?

Comment (themalkolm) [inline luigi/file.py]: Up to you. I talk sed and awk fluently so mkdirs is a least terrible thing I do in my work :D

**Comment by themalkolm** [comment]
TODO: s/mkdirs/makedirs/g

File: luigi/file.py (hunk 1 above)
Comment (themalkolm) [inline luigi/file.py]: Yeah, `mkdirs` came as a child of `mkdir(linux)` & `os.makedirs` :)

**Comment by themalkolm** [comment]
ping

**Pull request merged**